test/nodejs/route.test.js

```diff
@@ -0,0 +1,28 @@
+const test = require('tape');
+var config = require('./fixtures/basic_config');
+var valhalla = require('../../');
+var router = new valhalla.Actor(JSON.stringify(config));
+
+test('route: can get a route in Hershey', function(assert) {
+  var hersheyRequest = '{"locations":[{"lat":40.546115,"lon":-76.385076,"type":"break"}, {"lat":40.544232,"lon":-76.385752,"type":"break"}],"costing":"auto"}';
+  var route = JSON.parse(router.route(hersheyRequest));
+  assert.ok(route);
+  assert.equal(route['trip']['status_message'], 'Found route between points', 'route was found');
+  assert.equal(route['trip']['legs'].length, 1, 'Route has one leg');
+  assert.equal(route['trip']['legs'][0]['maneuvers'].length, 2, 'Leg has two maneuvers');
+  assert.end();
+});
+
+test('route: returns an error if no edges found', function(assert) {
+  var hersheyRequest = '{"locations":[{"lat":5,"lon":-76.385076,"type":"break"}, {"lat":40.544232,"lon":-76.385752,"type":"break"}],"costing":"auto"}';
+  assert.throws(() => { router.route(hersheyRequest) }, /No suitable edges near location/, 'Throws correct error');
+  assert.end();
+});
+
+test('route: returns an error if request format is wrong', function(assert) {
+  var hersheyRequest = '{"locations":[40.546115,-76.385076], [40.544232,"lon":-76.385752],"costing":"auto"}';
+  // TODO: we are not throwing a very useful error in this case - we should track this
+  // down and throw something a little more descriptive
+  assert.throws(() => { router.route(hersheyRequest) }, /std::exception/, 'Throws an error');
+  assert.end();
+});
```

Comment:
Do you mind explaining more what this does? And is there a way to check to make sure that the deletion occurred?

Reply:
`router = null;` deletes a reference to the object and if there no other references the garbage collector will delete the object. `global.gc();` enforces deletion of objects without references by the garbage collector. This requires `--expose-gc` argument.

Non-intrusive memory leaks checking can be really tricky. It is possible to use address sanitizer configured as
```
cmake .. -DENABLE_SANITIZER=On -DENABLE_SERVICES=Off
```
and running the test as
```
LD_PRELOAD=/usr/lib/x86_64-linux-gnu/libasan.so.4 node --expose-gc test/nodejs/route.test.js
```
but v8 keeps a lot of objects in memory at exit that look like memory leaks and it can be hard to detect automatically memory leak regressions. 

I think it is ok to check via coverage reports that `Actor::Destructor`  is called under specific conditions: 1/ no references to the actor object, 2/ garbage collection is forced.  For this it is enough to add the tests post tear down function that forces GC and add use  `node --expose-gc`

Related issue https://github.com/kevinkreiser/prime_server/issues/71

Reply:
okay makes sense, thanks!